src/Providers/components/ProviderRow.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useTranslation } from 'src/internal/i18n';
+import { ProviderResource } from 'src/internal/k8s';
+
+import { getMostSeriousCondition, getStatusType } from '@app/common/helpers';
+import { IStatusCondition } from '@app/queries/types';
+import { StatusIcon } from '@migtools/lib-ui';
+import { ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Popover } from '@patternfly/react-core';
+import { Td, Tr } from '@patternfly/react-table';
+
+import { NAME, NAMESPACE, READY, TYPE, URL } from './shared';
+import { RowProps } from './TableView';
+interface CellProps {
+  value: string;
+  resource: ProviderResource;
+  t(key: string): string;
+  kind: string;
+}
+const StatusCell = ({ value, resource, t }: CellProps) => {
+  return (
+    <Popover
+      hasAutoWidth
+      bodyContent={
+        <div>
+          {resource?.status?.conditions?.map((condition) => {
+            const severity = getMostSeriousCondition([
+              condition as IStatusCondition,
+            ]);
+            return (
+              <StatusIcon
+                key={condition.message}
+                status={getStatusType(severity)}
+                label={condition.message}
+              />
+            );
+          }) ?? 'No information'}
+        </div>
+      }
+    >
+      <Button variant="link" isInline aria-label={t(value)}>
+        <StatusIcon
+          status={value === 'Yes' ? 'Ok' : 'Error'}
+          label={t(value)}
+        />
+      </Button>
+    </Popover>
+  );
+};
+
+const TextCell = ({ value }: CellProps) => <>{value}</>;
+
+const ProviderLink = ({ value, resource, kind }: CellProps) => (
+  <ResourceLink
+    kind={kind}
+    name={value}
+    namespace={resource?.metadata?.namespace}
+  />
+);
+
+const cellCreator = {
+  [NAME]: ProviderLink,
+  [READY]: StatusCell,
+  [URL]: TextCell,
+  [TYPE]: TextCell,
+  [NAMESPACE]: ({ value }: CellProps) => (
+    <ResourceLink kind="Namespace" name={value} />
+  ),
+};
+
+const ProviderRow = (kind: string) =>
+  function ProviderRow({ columns, resource }: RowProps<ProviderResource>) {
+    const { t } = useTranslation();
+
+    return (
+      <Tr>
+        {columns.map(({ id, tKey, toValue }) => (
```

Comment:
```suggestion
        {columns.map(({ id, tKey, toValue,  cellCreator}) => (
```

Reply:
not sure about the name `cellCreator` , but the idea is to have the renderer optional part of the interface

Reply:
The idea is to keep `Field` interface data-only (at least as much as possible). This keeps it small and allows easy "flow" through components (via props) or storing( `useState` or local storage).
The extension point for rendering the **whole** entity is the `Row` prop (for tables).

Reply:
> The idea is to keep Field interface data-only (at least as much as possible)

methods are data ( [methods as data](https://developer.mozilla.org/en-US/docs/Glossary/First-class_Function) ) this property of JS can be useful when we want to attach different behavior using an interface, like we want here.

for example:
```ts
interface Animal {
  name: string;
  say: (something: string) => void;
}
 
const dog: Animal = {
  name: 'Dog',
  say: (something: string) => console.log(`barking: ${something}`)
}

const animals = [dog];
animals.map( (animal) => animal.say('hello') );
```

> This keeps it small and allows easy "flow" through components (via props)

we pass complex method arguments by reference, we can pass big data structures without the penalty for passing big chucks of data, passing a struct that holds all the information we need costs the same as passing an index and then reconstruct our data structure at the consumer.

e.g. doing:
```ts
type arr1 = { index: 1 | 2 | 3 };
type arr2 = { name: string};

const myArray: Array<arr1> = [{index: 1}, {index: 2}, {index: 3}];

const consume = (data: Array<arr1>) => {
  // Assume  data is a complex data structure including both methods and static data
  const myArrayData = { 1: {name: 'shoul'}, 2: {name: 'david'}, 3: {name: 'shlomo'}};
  
  return data.forEach( (item) => console.log( myArrayData[item['index']]['name'] ));
}

consume(myArray)
```
is the same as:
```ts
type arr1 = { index: number; name: string };

const myArray = [{index: 1, name: 'shoul'}, {index: 2, name: 'david'}, {index: 3, name: 'shlomo'}];

const consume = (data: Array<arr1>) => {
  return data.forEach( (item) => console.log( item.name ));
}

consume(myArray)
```

IMHO it's nice to have one array of cloumn definitions that hold all information, somethink like:
```ts
const columns: Column[] = [
  { label: 'Type', renderMethod: (rowFlatData) => Link(rowFlatData.type), ... },
  ...
];
```

Reply:
@rszwajko , so if we don't think of them as `columns`, but as `fieldMetadata` and `defaultFields` to render in table view, this makes more sense. IMHO we need to rename things to make sure a code reader will understand this are `fieldMetadata` that coupled with the `flatRowData` will help the view to get some semantics about how to read the data in the big struct.

Reply:
The newest version contains some of the suggested refactorings. 
The `fieldsMetadata` are constant and are used to initialize the `ProviderPage` (as `defaultFields`).
The `fields` used by the `ProvidersPage` are still view-independent but turn into `columns` when passed to a `TableView`.